enumerable_methods.rb

```diff
@@ -0,0 +1,63 @@
+# Project 2: Enumerable Methods
+
+module Enumerable
+  def my_each(&proc)
+    length.times do |i|
+      proc.call(self[i])
+    end
+    self
+  end
+
+  def my_each_with_index(&proc)
+    length.times do |item, _i|
+      proc.call(item)
+    end
+    self
+  end
+
+  def my_select(&proc)
+    selection = []
+    length.times do |i|
+      selection.push(self[i]) if proc.call(self[i])
+    end
+    selection
+  end
+
+  def my_all?(&proc)
+    length.times do |i|
+      return false if proc.call(self[i]) == false
+    end
+  end
+
+  def my_any?(&proc)
+    length.times do |i|
+      return true if proc.call(self[i]) == true
+    end
+  end
+
+  def my_none?(&proc)
+    length.times do |i|
+      return true if proc.call(self[i]) != true
+    end
+  end
+
+  def my_map(&proc)
+    my_each do |i|
+      self[i] = proc.call(self[i])
+    end
+    self
+  end
+
+  def my_inject(*arr)
+    init = arr.size.positive?
+    result = init ? arr[0] : self[0]
+    drop(init ? 0 : 1).my_each do |item|
+      result = proc.call(result, item)
+    end
+    result
+  end
```

Comment:
* [ ] should have working `my_inject` identical and returns the same thing as ruby's [`inject`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-inject). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_inject.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))

* [ ] my_inject when a symbol is specified combines each element of the collection by applying the symbol as a named method
  ```
    array = [1,5,8,7,9,5,5,55,5]
    p array.my_inject(:+) 
    # return this errors 
     # ArgumentError:
        tried to create Proc object without a block
      # ./enumerable_methods.rb:55:in `proc'
      # ./enumerable_methods.rb:55:in `block in my_inject'
      # ./enumerable_methods.rb:6:in `block in my_each'
      # ./enumerable_methods.rb:5:in `times'
      # ./enumerable_methods.rb:5:in `my_each'
      # ./enumerable_methods.rb:54:in `my_inject'
  ```